### ISSUE TITLE
function-component-definition should only test PascalCase identifiers

### ISSUE THREAD

**Issue by wmertens** [yannickcr/eslint-plugin-react#2778]
```js
function meep(props) {
  return <Meep />
}
```

Should not be changed by function-component-definition.

**Comment by ljharb** [comment]
While this is true, why would you have a function like this in the first place, that takes props and returns jsx and isn't named PascalCased?

**Comment by wmertens** [comment]
This would be a render function, which is passed to a component and gets called as part of the render phase

**Comment by ljharb** [comment]
Render functions don't take props.

**Comment by wmertens** [comment]
They might, and in #2771 the examples actually do

**Comment by ljharb** [comment]
They take *arguments* - props are *only* taken by elements/components.

**Comment by wmertens** [comment]
...ok, but they are indistinguishable in code?

```js
function Foo({bar}) { return <span>{bar}</span/> }
function renderFoo({bar}) { return <span>{bar}</span/> }
```

**Comment by ljharb** [comment]
Nope :-) but conceptually they're distinct, and you named your argument "props" which is a signal that it's a component, as is, taking a single destructured object argument - render props usually take multiple flat arguments.

**Comment by ljharb** [comment]
This seems answered; if not, we can reopen.
